storage: require min version file [draft]

RaduBerinde opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/97047)
<!-- Reviewable:end -->

RaduBerinde committed:
storage: require min version file

This change makes the min version file a requirement for opening an
existing store.

We have been writing this file out for at least two releases, so this
doesn't affect our compatibility story.

Release note: None
Epic: none
